backbone/src/main/java/org/researchstack/backbone/ui/step/layout/FormStepLayout.java

```diff
@@ -286,8 +288,19 @@ protected void onNextClicked()
         boolean isAnswerValid = isAnswerValid(true);
         if (isAnswerValid)
         {
-            updateAllQuestionSteps(false);
-            callbacks.onSaveStep(StepCallbacks.ACTION_NEXT, formStep, stepResult);
+            // WORKAROUND : Hide softkeyboard before transaction
+            InputMethodManager imm = (InputMethodManager) getContext().getSystemService(Context.INPUT_METHOD_SERVICE);
+            if (imm != null && container != null) {
+                imm.hideSoftInputFromWindow(container.getWindowToken(), 0);
+            }
+            final Handler handler = new Handler();
+            handler.postDelayed(new Runnable() {
+                @Override
+                public void run() {
+                    updateAllQuestionSteps(false);
+                    callbacks.onSaveStep(StepCallbacks.ACTION_NEXT, formStep, stepResult);
+                }
+            }, 100);
```

Comment:
Wasn't this already in somewhere else?

Reply:
It is, and testing the app I found was needed on another part of the app :(